h/static/scripts/controllers/form-input-controller.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var inherits = require('inherits');
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for form input fields.
+ */
+function FormInputController(element) {
```

Comment:
Perhaps I'm missing something, but why go to all the trouble to add `Controller#trigger` and `Controller#on`, when you could just use event delegation in `FormController` to set up the same handlers?